src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.component.html

```diff
@@ -0,0 +1,72 @@
+<h1>
+    <!--Default IDE Settings-->
+    {{ 'artemisApp.userSettings.idePreferences' | artemisTranslate }}
+</h1>
+<div class="list-group d-block">
+    <div class="list-group-item pb-3">
+        <dt>
+            <span jhiTranslate="artemisApp.userSettings.idePreferencesPage.description"> </span>
+        </dt>
+        <dt class="py-2">
+            {{ 'artemisApp.userSettings.idePreferencesPage.default' | artemisTranslate }}
+        </dt>
+        <div class="btn-group">
+            @for (ide of PREDEFINED_IDE; track ide) {
+                <div
+                    class="btn btn-default"
+                    [ngClass]="{
+                        'btn-primary selected': isIdeOfProgrammingLanguage(ProgrammingLanguage.EMPTY, ide),
+                        'btn-default': !isIdeOfProgrammingLanguage(ProgrammingLanguage.EMPTY, ide),
+                    }"
+                    (click)="changeIde(ProgrammingLanguage.EMPTY, ide)"
+                >
+                    {{ ide.name }}
+                </div>
+            }
+        </div>
+    </div>
+    <!-- Display each customized Programming Language -->
+    @for (programmingLanguage of getCustomizedProgrammingLanguages(); track programmingLanguage) {
```

Comment:
Don't call methods in the template as the change detection cannot handle them and calls them over and over again

Reply:
what is the best way to do it then?